gym_taxi_goal/taxi_goal.py

```diff
@@ -0,0 +1,54 @@
+import gym
+from gym.envs.toy_text.taxi import TaxiEnv
+
+STATE_MOVE_TO_PASSENGER = 1
+STATE_PICKUP_PASSENGER = 2
+STATE_MOVE_TO_DEST = 3
+STATE_DROPOFF_PASSENGER = 4
+STATE_OVER = 5
+
+
+class TaxiGoalEnv(TaxiEnv):
+    def __init__(self):
+        TaxiEnv.__init__(self)
+        self.goal_generator_state = STATE_OVER
+
+    def get_goal_state(self):
+        """
+        Goal generator func.
+        1. Move to the passenger location.
+        2. Pick up the passenger.
+        3. Move to destination location.
+        4. Drop off the passenger.
+        :return:
+        """
+        decoded = list(self.decode(self.s))
```

Comment:
I'd would rather do it like `taxirow, taxicol, passloc, destidx = list(self.decode(self.s))`